Sources/BraintreeCore/BTAPIClient.swift

```diff
@@ -93,23 +93,20 @@ import Foundation
     /// cached on subsequent calls for better performance.
     @_documentation(visibility: private)
     public func fetchOrReturnRemoteConfiguration(_ completion: @escaping (BTConfiguration?, Error?) -> Void) {
-        configurationLoader.getConfig { [weak self] configuration, error in
-            guard let self else {
-                completion(nil, BTAPIClientError.deallocated)
-                return
-            }
-            
-            if let error {
+        // TODO: - Consider updating all feature clients to use async version of this method?
+
+        Task { @MainActor in
```

Comment:
I wonder if we remove this and just run the setupHTTPCredentials and completions on MainActor, we wouldn't need @MainActor annotation on fetchConfiguration. I am concerned about potential latency issues of sendingAnalytics on main.